lib/seahorn/BvOpSem2.cc

```diff
@@ -555,6 +579,53 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
     setValue(*CS.getInstruction(), res);
   }
 
+  void visitAssertIf(CallSite CS) {
+    // NOTE: assert.if.not is not supported
+    if (VacuityCheckOpt == VacCheckOptions::NONE) {
+      return;
+    }
+    auto I = CS.getInstruction();
+    Expr ante = lookup(*CS.getArgument(0));
+    tribool anteRes = solveWithConstraints(ante);
+    // if ante is unsat then report false and bail out
+    if (!anteRes) {
+      LOG("opsem", ERR << "Antecedent is unsat/unknown: " << *I << "\n");
+      return; // return early since conseq is unreachable
+    }
+    if (VacuityCheckOpt == VacCheckOptions::ANTE) {
+      return; // don't need to process conseq
+    }
+    Expr conseq = lookup(*CS.getArgument(1));
+    Expr e = boolop::land(ante, boolop::lneg(conseq));
+    tribool conseqRes = solveWithConstraints(e);
+    if (conseqRes) {
+      LOG("opsem", ERR << "Consequent is sat/unknown: " << *I << "\n");
+    }
+  }
+
+  tribool solveWithConstraints(const Expr &solveFor) const {
+    // OPTIMIZE: Solver need not be reset
+    m_ctx.solver().reset();
+    ExprVector conds(m_ctx.side());
+    conds.push_back(m_ctx.getPathCond());
+    for (auto e : conds) {
+      m_ctx.solver().assertExpr(e);
+    }
+    m_ctx.solver().assertExpr(solveFor);
+
+    boost::tribool res = m_ctx.solver().solve();
+    return res;
+  }
+
+  void visitAssertStmt(CallSite CS) {
+    // This is called when UnifyAssumes does not replace
+    // verifier.assert with sea.assert.if e.g.
+    // when no assumes is found, as a workaround do the following
+    // TODO: add logic to delegate to sea.assert.if(true, conseq)
```

Comment:
hope this todo is marked in the project as an issue and will not be hidden in here.

Reply:
done